logs/parse.go

```diff
@@ -306,7 +331,22 @@ func (lp *LogParser) GetOccurredAt(timePart string) time.Time {
 	return ts
 }
 
-func parseSyslogLine(line string, tz *time.Location) (logLine state.LogLine, ok bool) {
+var UserRegexp = `(\S*)`                              // %u
+var DbRegexp = `(\S*)`                                // %d
+var AppInsideBracketsRegexp = `(\[unknown\]|[^,\]]*)` // %a
+var PidRegexp = `(\d+)`                               // %p
+
+var SyslogSequenceAndSplitRegexp = `(\[[\d-]+\])?`
+var LevelAndContentRegexp = `(\w+):\s+(.*\n?)$`
+var LogPrefixNoTimestampUserDatabaseAppRegexp = regexp.MustCompile(`(?s)^\[user=` + UserRegexp + `,db=` + DbRegexp + `,app=` + AppInsideBracketsRegexp + `\] ` + LevelAndContentRegexp)
+
+var RsyslogLevelAndContentRegexp = `(?:(\w+):\s+)?(.*\n?)$`
+var RsyslogTimeRegexp = `(\w+\s+\d+ \d{2}:\d{2}:\d{2})`
+var RsyslogHostnameRegxp = `(\S+)`
+var RsyslogProcessNameRegexp = `(\w+)`
+var RsyslogRegexp = regexp.MustCompile(`^` + RsyslogTimeRegexp + ` ` + RsyslogHostnameRegxp + ` ` + RsyslogProcessNameRegexp + `\[` + PidRegexp + `\]: ` + SyslogSequenceAndSplitRegexp + ` ` + RsyslogLevelAndContentRegexp)
+
+func (lp *LogParser) parseSyslogLine(line string) (logLine state.LogLine, ok bool) {
```

Comment:
Similarly, made this a LogParser method instead of a standalone function in order to simplify GetOccurredAt handling.